test/units/plugins/lookup/test_password.py

```diff
@@ -20,117 +20,349 @@
+    dict(term=u'/path/with/embedded spaces and/file chars=abc=def',
+         filename=u'/path/with/embedded spaces and/file',
+         params=dict(length=password.DEFAULT_LENGTH, encrypt=None, chars=[u'abc=def']),
+         candidate_chars=u'abc=def',
+         ),
+    dict(term=u'/path/with/equals/cn=com.ansible chars=abc=def',
+         filename=u'/path/with/equals/cn=com.ansible',
+         params=dict(length=password.DEFAULT_LENGTH, encrypt=None, chars=[u'abc=def']),
+         candidate_chars=u'abc=def',
+         ),
+    dict(term=u'/path/with/unicode/くらとみ/file chars=くらとみ',
+         filename=u'/path/with/unicode/くらとみ/file',
+         params=dict(length=password.DEFAULT_LENGTH, encrypt=None, chars=[u'くらとみ']),
+         candidate_chars=u'くらとみ',
+         ),
+)
+
+
+class TestLookupModule(unittest.TestCase):
+    @patch.object(PluginLoader, '_get_paths')
+    @patch('ansible.plugins.lookup.password._create_password_file')
+    @patch('ansible.plugins.lookup.password._write_password_file')
+    def test(self, mock_get_paths, mock_create_pf, mock_write_file):
+        mock_get_paths.return_value = ['/path/one', '/path/two', '/path/three']
+        fake_loader = DictDataLoader({'/path/to/somewhere':'sdfsdf'})
+        password_lookup = password.LookupModule(loader=fake_loader)
+        results = password_lookup.run([u'/path/to/somewhere'],
+                                  None)
+
+        # FIXME: assert something useful
+        for result in results:
+            self.assertIsInstance(result, text_type)
+
+    @patch.object(PluginLoader, '_get_paths')
+    @patch('ansible.plugins.lookup.password._create_password_file')
+    @patch('ansible.plugins.lookup.password._write_password_file')
+    def test_encrypt(self, mock_get_paths, mock_create_pf, mock_write_file):
+        mock_get_paths.return_value = ['/path/one', '/path/two', '/path/three']
+        fake_loader = DictDataLoader({'/path/to/somewhere':'sdfsdf'})
+        password_lookup = password.LookupModule(loader=fake_loader)
+        results = password_lookup.run([u'/path/to/somewhere encrypt=pbkdf2_sha256'],
+                                      None)
+
+        # FIXME: assert something useful
+        for result in results:
+            self.assertIsInstance(result, text_type)
+
+    @patch.object(PluginLoader, '_get_paths')
+    @patch('ansible.plugins.lookup.password._create_password_file')
+    @patch('ansible.plugins.lookup.password._write_password_file')
+    def test_only_a(self, mock_get_paths, mock_create_pf, mock_write_file):
+        mock_get_paths.return_value = ['/path/one', '/path/two', '/path/three']
+        fake_loader = DictDataLoader({'/path/to/somewhere':'sdfsdf'})
+        password_lookup = password.LookupModule(loader=fake_loader)
+        results = password_lookup.run([u'/path/to/somewhere chars=a'],
+                                  None)
+        for result in results:
+            for char in result:
+                self.assertEqual(char, u'a')
```

Comment:
You should be able to know the length too (either the default or specify it) and then you don't have to loop through each character in result.